### PR TITLE
CI: speedup 'Check Asset Sizes' job (take 2)

### PR DESCRIPTION
**Pull request by kalessil**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Spinoff of https://github.com/woocommerce/woocommerce/pull/49793, where we actualize CI-optimized install/build commands (same as monorepo setup action used under the hood).

This and previous speedup attempts should give us around a 1-1.5 minute boost and allow us to stay within 10-minute goal.
Practical results, though, are to be evaluated from job execution times stats.

### Testing

- `Create issues for flaky tests` job details (under `CI`) should say it's been skipped (as no test-jobs were executed)
- `Check Asset Sizes` during assets build steps should mention `Scope: 34 of 49 workspace projects` (reduced build scope)

### CODE REVIEW

File: .github/workflows/pr-assess-bundle-size.yml [hunk 1]
```diff
@@ -50,6 +50,8 @@ jobs:
               with:
                   repo-token: '${{ secrets.GITHUB_TOKEN }}'
                   pattern: './{packages/js/!(*e2e*|*internal*|*test*|*plugin*|*create*),plugins/woocommerce-blocks}/{build,build-style}/**/*.{js,css}'
+                  install-script: 'pnpm install --filter="@woocommerce/plugin-woocommerce..." --frozen-lockfile --config.dedupe-peer-dependents=false'
+                  build-script: '--filter="@woocommerce/plugin-woocommerce" build'
```
Comment (kalessil) [inline .github/workflows/pr-assess-bundle-size.yml:56]: Not directly related, but `Create issues for flaky tests` job should not be triggered for this PR (TBD: incorporate` if: ${{ needs.project-jobs.outputs.test-jobs != '[]' }}`).

Also, passing by CI workflow: `Evaluate Project Job Statuses` shouldn't start if workflow cancellation is in progress.

Comment (kalessil) [reply .github/workflows/pr-assess-bundle-size.yml:56]: TBD: add `BROWSERSLIST_IGNORE_OLD_DATA: true` to the job environment.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -207,7 +207,7 @@ jobs:
         'project-lint-jobs',
         'project-test-jobs',
       ]
-    if: ${{ always() && github.event_name == 'pull_request' }}
+    if: ${{ !cancelled() && github.event_name == 'pull_request' }}
```
Comment (kalessil) [inline .github/workflows/ci.yml:210]: Ensures `Evaluate Project Job Statuses` will not start if workflow cancellation is requested.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -330,8 +330,12 @@ jobs:
   
   report-flaky-tests:
     name: 'Create issues for flaky tests'
-    if: ${{ !cancelled() && ! github.event.pull_request.head.repo.fork }}
-    needs: [ 'project-test-jobs' ]
+    if: ${{ !cancelled() && ! github.event.pull_request.head.repo.fork && needs.project-jobs.outputs.test-jobs != '[]' }}
```
Comment (kalessil) [inline .github/workflows/ci.yml:333]: Ensures `Create issues for flaky tests` will not start if no test jobs was planned.

**Review by adimoldovan — APPROVED** [review]
LGTM!

**Pull request merged**